UriGenerator.php

```diff
@@ -54,7 +55,18 @@ public function generateUri(UriContext $uriContext)
         $tokenProviderConfigs = $metadata->getTokenProviders();
 
         $tokens = array();
-        foreach ($tokenProviderConfigs as $name => $options) {
+        preg_match_all('/{(.*?)}/', $metadata->getUriSchema(), $matches);
```

Comment:
creation is not something that happens at a high rate normally, so i am not worried about performance here. good sanity checks have priority over performance imho.